File: donghyo/Programmers/mo2gosa.py
from collections import deque
def solution(answers):
    answer = []
    first_Student = [1, 2, 3, 4, 5]
    second_Student = [2, 1, 2, 3, 2, 4, 2, 5]
    third_Student = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    score = [0] * 3

    for i in range(len(answers)):
        if answers[i] == first_Student[i % 5]:
            score[0] += 1
        if answers[i] == second_Student[i % 8]:
            score[1] += 1
        if answers[i] == third_Student[i % 10]:
            score[2] += 1

    max_Student = max(score[0], score[1], score[2])

    for i in range(3):
        if score[i] == max_Student:
            answer.append(i+1)

    print(answer)
    return answer

solution([1,3,2,4,2])

File: donghyo/Programmers/mo2gosa.js
function solution(answers) {
    var answer = [];
    let supo1 = [1, 2, 3, 4, 5];
    let supo2 = [2, 1, 2, 3, 2, 4, 2, 5];
    let supo3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
    let score = [0, 0, 0];

    for (let i = 0; i <= answers.length; i++) {
        if (answers[i] == supo1[i % 5]) {
            score[0] += 1;
        }
        if (answers[i] == supo2[i % 8]) {
            score[1] += 1;
        }
        if (answers[i] == supo3[i % 10]) {
            score[2] += 1;
        }
    }

    let max_student = Math.max(...score)
    console.log(score)

    for (let i = 0; i <= 3; i++) {
        if (score[i] == max_student) {
            answer.push(i + 1);
        }
    }
    console.log(answer)

    return answer;
}
solution([1, 3, 2, 4, 2]);